src/data/finset.lean

```diff
@@ -980,6 +980,9 @@ range_succ
 
 @[simp] theorem not_mem_range_self : n ∉ range n := not_mem_range_self
 
+theorem self_mem_range_succ (n : ℕ) : n ∈ range (n + 1) :=
+by simp only [succ_pos', lt_add_iff_pos_right, mem_range]
```

Comment:
Ideally lemmas about `finset.range` have corresponding lemmas for `multiset.range` and `list.range` (and ideally each is proved from the previous one).

It's not necessary that you do this in this PR, but just worth being aware we accumulate some "mismatch debt" between the files if they only get added part way up the hierarchy.